docs/src/tutorial/embedding.rst

```diff
@@ -58,30 +57,123 @@ to export it as a linker symbol that can be used by other C files, which in this
 
 .. literalinclude:: ../../examples/tutorial/embedding/embedded_main.c
+    :caption: embedded_main.c
     :linenos:
     :language: c
 
 (Adapted from the `CPython documentation
 <https://docs.python.org/3/extending/extending.html#the-module-s-method-table-and-initialization-function>`_.)
 
 Instead of writing such a ``main()`` function yourself, you can also let
-Cython generate one into your module's C file with the ``cython --embed``
-option.  Or use the
-`cython_freeze <https://github.com/cython/cython/blob/master/bin/cython_freeze>`_
-script to embed multiple modules.  See the
-`embedding demo program <https://github.com/cython/cython/tree/master/Demos/embed>`_
-for a complete example setup.
-
-Be aware that your application will not contain any external dependencies that
-you use (including Python standard library modules) and so may not be truly portable.
-If you want to generate a portable application we recommend using a specialized
-tool (e.g. `PyInstaller <https://pyinstaller.org/en/stable/>`_
-or `cx_freeze <https://cx-freeze.readthedocs.io/en/latest/index.html>`_) to find and
-bundle these dependencies.
+Cython generate one into your module's C file with the ``cython --embed`` option.
+
+When using ``--embed``, Cython generates the C implementation file (e.g., ``embedded.c``) along with a header file (e.g., ``embedded.h``).
+
+The generated ``embedded.c`` file contains a ``__Pyx_main()`` function. This function initializes the Python interpreter and can be modified to call public functions exported by your Cython module (such as ``say_hello_from_python()``).
+
+The full process for compiling and running is as follows:
+
+.. code-block:: bash
+
+    gcc -c embedded.c $(python3-config --cflags) -o embedded.o
+    gcc embedded.o $(python3-config --ldflags --embed) -o embedded
+
+After compilation, the ``embedded`` executable is created. Running it will execute the content of ``__Pyx_main()``:
+
+.. code-block:: bash
+
+    $ ./embedded
+    Hello from Python!
+
+
+Embedding multiple modules
+==========================
+
+To statically link multiple Cython modules into a single executable, you must use the
+``cython --embed-modules`` option alongside the ``cython --embed`` option.
+This option enables the necessary C-level initialisation calls (via :c:func:`PyImport_AppendInittab`)
+for all listed modules within the main program's C code.
+
+The argument is a comma-separated list of the module names you wish to embed:
+
+.. code-block:: bash
+
+    cython --embed --embed-modules=lcmath,another_module main_script.pyx
+
+To illustrate how it works, let's assume we have the following files:
+
+.. code-block:: cython
+   :caption: lcmath.pyx
+
+    def add(a, b):
+        return a + b
+
+    cdef int sub(int a, int b):
+        return a - b
+
+.. code-block:: cython
+   :caption: lcmath.pxd
+
+    cdef int sub(int a, int b)
+
+.. code-block:: cython
+   :caption: combinatorics.pyx
```

Comment:
You might want to add a simple `__main__` section here to illustrate the use of the C `main()` function. Just a simple "print(f"… {sub_one(…)})" or something like that. Otherwise, starting this program will do nothing.